integration_test_suite/integration_tests/src/test/java/io/prometheus/jmx/test/http/ssl/SSLWithTrustStoreAndClientAuth.java

```diff
@@ -0,0 +1,277 @@
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+import org.assertj.core.util.Strings;
+import org.opentest4j.AssertionFailedError;
+import org.testcontainers.containers.Network;
+import org.verifyica.api.ArgumentContext;
+import org.verifyica.api.ClassContext;
+import org.verifyica.api.Trap;
+import org.verifyica.api.Verifyica;
+
+public class SSLWithTrustStoreAndClientAuth {
+
+    private static final String BASE_URL = "https://localhost";
+
+    @Verifyica.ArgumentSupplier() // not parallel as the static HttpsURLConnection
+    // defaultSSLSocketFactory is manipulated
+    public static Stream<ExporterTestEnvironment> arguments() {
+        // Filter Java versions that don't support the PKCS12 keystore
+        // format or don't support the required TLS cipher suites
+        return ExporterTestEnvironment.createExporterTestEnvironments()
+                .filter(new PKCS12KeyStoreExporterTestEnvironmentFilter())
+                .map(exporterTestEnvironment -> exporterTestEnvironment.setBaseUrl(BASE_URL));
+    }
+
+    @Verifyica.Prepare
+    public static void prepare(ClassContext classContext) {
+        TestSupport.getOrCreateNetwork(classContext);
+    }
+
+    @Verifyica.BeforeAll
+    public void beforeAll(ArgumentContext argumentContext) {
+        Class<?> testClass = argumentContext.classContext().testClass();
+        Network network = TestSupport.getOrCreateNetwork(argumentContext);
+        TestSupport.initializeExporterTestEnvironment(argumentContext, network, testClass);
+    }
+
+    private SSLContext initSSLContextForClientAuth(JmxExporterMode mode) {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+
+            // to verify cert auth with existing test pki resources, use self-signed server cert as
+            // client cert and source of trust
+            final String type = "PKCS12";
+            final char[] password = "changeit".toCharArray();
+            final String keyStoreResource =
+                    Strings.formatIfArgs(
+                            "%s/%s/localhost.pkcs12",
+                            this.getClass().getSimpleName(), mode.toString());
+            KeyStore keyStore = KeyStore.getInstance(type);
+            try (InputStream inputStream = this.getClass().getResourceAsStream(keyStoreResource)) {
+                keyStore.load(inputStream, password);
+            }
+            KeyManagerFactory km =
+                    KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            km.init(keyStore, password);
+            TrustManagerFactory tm =
+                    TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tm.init(keyStore);
+
+            sslContext.init(
+                    km.getKeyManagers(), tm.getTrustManagers(), new java.security.SecureRandom());
+
+            return sslContext;
+        } catch (IOException
```

Comment:
Why are we catching exceptions and logging them?

Reply:
should not be, good catch. thanks.